tgui/packages/tgui/interfaces/GasCanister/Detonator.js

```diff
@@ -0,0 +1,188 @@
+import { Fragment } from 'inferno';
+import { Button, LabeledList, Section, Box, Flex, Divider } from '../../components';
+import { DetonatorTimer } from './DetonatorTimer';
+
+export const Detonator = props => {
+  const {
+    detonator,
+    detonatorAttachments,
+    onToggleAnchor,
+    onToggleSafety,
+    onWireInteract,
+    onPrimeDetonator,
+    onTriggerActivate,
+    onSetTimer,
+  } = props;
+
+  return (
+    <Section
+      title="Detonator">
+      <DetonatorWires
+        detonator={detonator}
+        onWireInteract={onWireInteract}
+        onSetTimer={onSetTimer} />
+      <Divider />
+      <DetonatorUtility
+        detonator={detonator}
+        detonatorAttachments={detonatorAttachments}
+        onToggleAnchor={onToggleAnchor}
+        onToggleSafety={onToggleSafety}
+        onPrimeDetonator={onPrimeDetonator}
+        onTriggerActivate={onTriggerActivate} />
+    </Section>
+  );
+};
+
+const DetonatorWires = props => {
+  const {
+    detonator: {
+      wireNames,
+      wireStatus,
+      time,
+      isPrimed,
+    },
+    onWireInteract,
+    onSetTimer,
+  } = props;
+
+  return (
+    <Flex>
+      <Flex.Item>
+        <LabeledList>
+          { wireNames.map((entry, i) => (
+            <LabeledList.Item
+              key={entry + i}
```

Comment:
As it stands, no, not really. My thought was that, in the probably unlikely scenario that the wire names are changed on the backend (so that they're no longer unique), then the client doesn't break.